banzai/tasks.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from banzai import dbs
+from banzai import stages
+from banzai import main
+from banzai.utils import date_utils
+import banzai
+import logging
+import os
+from celery import Celery
+
+
+logger = logging.getLogger('banzai')
+
+app = Celery('tasks')
+app.config_from_object('banzai.settings')
+
+
+@app.task(bind=True, max_retries=3, default_retry_delay=3 * 60)
+def reduce_preview_image(path, pipeline_context):
+    if dbs.need_to_make_preview(path, db_address=pipeline_context.db_address,
+                                max_tries=pipeline_context.max_preview_tries):
+        stages_to_do = stages.get_stages_todo()
+
+        logging_tags = {'tags': {'filename': os.path.basename(path)}}
+        logger.info('Running preview reduction on {}'.format(path), extra=logging_tags)
+        pipeline_context.filename = os.path.basename(path)
+        pipeline_context.raw_path = os.path.dirname(path)
+
+        # Increment the number of tries for this file
+        dbs.increment_preview_try_number(path, db_address=pipeline_context.db_address)
```

Comment:
Just FYI you have access to self.request.retries (int number of times this task as retried) and self.retry() which re enqueues the task (takes a few helpful arguments as well)